### PR TITLE
Support nested structs (#1)

### PR DESCRIPTION
**Pull request by etloreq**

### CODE REVIEW

File: schema.go [hunk 1]
```diff
@@ -231,9 +258,49 @@ func compare(t reflect.Type, src map[string]interface{}, opts *CompareOpts, resu
...
+	// Remeber count of fields to check if new errors occured.
+	mismatchCount := len(results.MismatchedFields)
+	missingCount := len(results.MissingFields)
+
+	compare(t, src, opts, results)
+
+	// If there were new mismatched fields, add the current field name to their path.
+	if mismatchCount != len(results.MismatchedFields) {
+		for mi := mismatchCount; mi < len(results.MismatchedFields); mi++ {
+			results.MismatchedFields[mi].Path = append([]string{fieldName}, results.MismatchedFields[mi].Path...)
```
Comment (Kangaroux) [inline schema.go:294]: Is this backwards?

```go
results.MismatchedFields[mi].Path = append(results.MismatchedFields[mi].Path, fieldName)
```

File: schema.go [hunk 2]
```diff
@@ -231,9 +258,49 @@ func compare(t reflect.Type, src map[string]interface{}, opts *CompareOpts, resu
...
+		}
+	}
+
+	// If there were new missing fields, add the current field name to their path.
+	if missingCount != len(results.MissingFields) {
+		for mi := missingCount; mi < len(results.MissingFields); mi++ {
+			results.MissingFields[mi].Path = append([]string{fieldName}, results.MissingFields[mi].Path...)
```
Comment (Kangaroux) [inline schema.go:301]: Same here

File: schema.go [hunk 3]
```diff
@@ -231,9 +258,49 @@ func compare(t reflect.Type, src map[string]interface{}, opts *CompareOpts, resu
...
+		if shouldCheckNested {
+			nested := src[fieldName].(map[string]interface{})
+			nestedType := f.Type
+			if f.Type.Kind() == reflect.Ptr {
+				nestedType = nestedType.Elem()
+			}
+
+			checkNestedFields(nestedType, fieldName, nested, opts, results)
+
+			continue
```
Comment (Kangaroux) [inline schema.go]: Can remove this `continue`

File: lang.go [hunk 4]
```diff
@@ -74,3 +74,14 @@ func TypeNameWithArticle(t string) string {
 		return "a " + t
 	}
 }
+
+// FieldNameWithPath returns the field name incliding the path in the following format: parent1.parent2.name
```
Comment (Kangaroux) [inline lang.go]: Typo (incliding -> including)

File: schema.go [hunk 5]
```diff
@@ -231,9 +258,49 @@ func compare(t reflect.Type, src map[string]interface{}, opts *CompareOpts, resu
...
+		}
+	}
+}
+
+func checkNestedFields(t reflect.Type, fieldName string, src map[string]interface{}, opts *CompareOpts, results *CompareResults) {
+	// Remeber count of fields to check if new errors occured.
```
Comment (Kangaroux) [inline schema.go]: Typo (remeber -> remember)

**Review by Kangaroux — CHANGES_REQUESTED** [review]
Nice work!

File: schema.go (hunk 1 above)
Comment (etloreq) [reply schema.go:294]: We add parents on recursion ascend, so we are getting them in reverse order. I add them at the beginning of the Path to keep the Path in direct order
If the order in the Path doesn't matter I can change as you suggest

File: lang.go (hunk 4 above)
Comment (etloreq) [reply lang.go]: Fixed

File: schema.go (hunk 5 above)
Comment (etloreq) [reply schema.go]: Fixed

File: schema.go (hunk 3 above)
Comment (etloreq) [reply schema.go]: Fixed

**Pull request merged**